app/models/bank_account.rb

```diff
@@ -64,4 +62,18 @@ def set_initial_residue
   def to_s
     "#{name.truncate(30)} (#{Money::Currency.new(currency).symbol})"
   end
+
+  class << self
+    def grouped_by_currency(currency)
+      currencies = Currency.ordered(currency)
+
+      grouped = []
```

Comment:
`BankAccount.all.index_by(&:currency)` returns only 1 bank account for every currency. I think it rewrites keys values